### DISCUSSION TITLE
"Inertia-less" ships, like Vell-Os from EV Nova

### DISCUSSION THREAD

**Discussion by Ivo-Pivo** [endless-sky/endless-sky#8149] [Balance, Mechanics, and Interactions]
I recently picked this incredible game up after being disappointed that my old EVN installation had broken (an experience I think is shared between many of us). Apart from the fact that ES strikes me as an objectively good game, I was also impressed and intrigued by the fact that it's open source, so here I am, getting in touch with you developers:

I'm surprised to find no discussion of any kind regarding "inertia-less" ships. For context, these ships would always fly the direction they were pointed, i.e their velocity vector was always equal to their heading. A bit like having a "keel", like boats. This would give a very unique play-style of hit-and-runs. For additional (lore) context, these ships weren't actually ships in the first place, rather they were constructs of the pilots mind, which somewhat justifies the un-natural behavior. I saw some discussion about a "solar sail" mechanic which I think could have some synergy.

I can imagine this mechanic being disregarded as unfeasible or otherwise unwanted, but the lack of discussion implies that it has simply been overlooked entirely. Do you guys think there is a place for this kind of ship/propulsion in ES?

**Comment by Zitchas** [comment]
While I wasn't fond of flying that style of ship myself, I found it very interesting to have a greater variety of movement styles within the game. 

That being said, while it's not as absolute as the inertia-less movement, I've found that ES's "drag" system of physics does have some similarities. (This is seen in the fact that if you hold down your accelerate key in any particular direction, eventually your velocity vector will be entirely aligned with the direction you are going, with whatever other motion you had nullified. This isn't how physics works, but it is how things are in ES currently.) It's kind of a very slow intertialess setup.

**Comment by vitalchip** [comment]
It's been discussed before. May 2021 if the date on this clip is correct. I seem to remember there was some confusion or disagreement about how the ships flew but this was the best fit.

The ship only moves under thrust, stopping instantly when the key is released and it has no drift at all, just moves wherever the nose is pointed.

https://user-images.githubusercontent.com/65342577/212916772-bae49aa6-c77b-417b-81ff-f8109b5bfce5.mov

**Reply by Zitchas** [reply]
Yes, that looks like it.

Probably as a QoL thing, in EVN the inertialess movement actually changed how the controls worked. Hitting accelerate actually acted like pushing forward the thrust lever on an airplane. As in, each time you hit it, your thrust & speed went up. And stayed up, even after you released the key. And would continue firing until you hit the decelerate button, which acted the same way: Each tap reduced the thrust until it reached zero. I recall seeing people on the forums calling it "tractor controls" because it kind of felt like a tractor in some regards.

**Reply by Ivo-Pivo** [reply]
Cool, thanks for finding that! Clearly my searching ability failed me. And in addition to what Zitchas said, in general I remember the Vell-Os ships having quite high turn rates which also made the navigation more bearable.

**Reply by Ivo-Pivo** [reply]
I do think this mode of movement has some potential, perhaps for a new faction/race, but I would have to keep tinkering with  that idea for a bit. For now, I'm just satisfied to find some sort of previous discussion on the subject. Where did you find that clip, and is there any more context for it?

**Reply by vitalchip** [reply]
It's my clip. I posted it on Discord back then during a discussion. I still had it on my machine.

**Reply by vitalchip** [reply]
The thing with having to tap and release the key to build the thrust is possible too, don't recall anyone mentioning it before. It's really just a burst of thrust mechanic though and a haard limit at zero wrt to reverse as I was told the ships couldn't move backwards (which makes sense if the velocity is always as the ship faces.

We could catch the actual key presses in Engine but if something like this were introduced I think it would be better to invoke a 'thrust sanitizer' in Ai to deal with it on a ship by ship basis, that way Ai ships would need less special handling.
